addons/payment/models/account_journal.py

```diff
@@ -69,3 +69,11 @@ def _compute_available_payment_method_ids(self):
                     to_remove.append(payment_method.id)
 
                 journal.available_payment_method_ids = [Command.unlink(payment_method) for payment_method in to_remove]
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_to_payment_acquirer(self):
+        linked_payment_acquirers = self.env['payment.acquirer'].sudo().search([]).filtered_domain(
+                [('journal_id', 'in', self.ids)])
+        if linked_payment_acquirers:
+            raise UserError(_("You can't delete a journal that is linked to an acquirer.\n"
+                              "Linked acquirer(s): %s", ', '.join(a.display_name for a in linked_payment_acquirers)))
```

Comment:
Please rewrite as follow to:
- be easier to read
- ignore disabled acquirers
- rephrase the user error
```suggestion
    @api.ondelete(at_uninstall=False)
    def _unlink_except_linked_to_payment_acquirer(self):
        linked_acquirers = self.env['payment.acquirer'].sudo().search([]).filtered(
            lambda acq: acq.journal_id.id in self.ids and acq.state != 'disabled'
        )
        if linked_acquirers:
            raise UserError(_(
                "You must first deactivate a payment acquirer before deleting its journal.\n"
                "Linked acquirer(s): %s", ', '.join(acq.display_name for acq in linked_acquirers)
            ))
```